src/main/java/io/spokestack/spokestack/asr/SpokestackCloudClient.java

```diff
@@ -0,0 +1,294 @@
+package io.spokestack.spokestack.asr;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import io.spokestack.spokestack.util.Crypto;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.WebSocket;
+import okhttp3.WebSocketListener;
+import okio.ByteString;
+import org.jetbrains.annotations.NotNull;
+
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Spokestack cloud speech client.
+ *
+ * <p>
+ * This client encapsulates the websocket logic used to communicate with
+ * Spokestack's cloud-based ASR service.
+ * </p>
+ */
+public class SpokestackCloudClient {
+    private static final int MESSAGE_BUFFER_SIZE = 1500;
```

Comment:
I think this buffer size will cause network packet fragmentation (not a huge deal) and audio frame fragmentation, which will delay parts of the frames that straddle the buffer boundary from reaching the server until the buffer can be filled again
From my local network, I get a path MTU to the api service of 1500 bytes, but with TLS and websocket overhead, the largest buffer I can send without fragmentation is 1411 bytes
It would probably make more sense to set the buffer size as a multiple of the audio frame size, so something like 1280 bytes would probably avoid packet fragmentation on most networks and it would avoid audio frame fragmentation, while sending mostly full packets

Reply:
That makes sense; I'll adjust.